src/main/java/com/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -33,10 +34,27 @@ public List<BookResponseDto> findAll() {
     }
 
     @Override
-    public BookResponseDto findById(Long id) {
+    public Optional<BookResponseDto> findById(Long id) {
         return bookRepository.findById(id)
```

Comment:
throw EntityNotFoundException if bookRepository returns empty Optional

Reply:
In this way?

Reply:
yes